bootstrap fails with ocaml 4.10

I think this used to work not long ago, now dune.git#master and ocaml.git#4.10 fail like this:

````
[  126s] + cd ocaml-dune-2.0
[  126s] ++ ocamlc -where
[  126s] + ocaml configure.ml --libdir=/usr/lib64/ocaml
[  126s] + /usr/bin/make -O -j16 V=1 VERBOSE=1 release-with-libraries
[  126s] ocaml bootstrap.ml
[  126s] ocamlc -output-complete-obj -w -24 -g -o .duneboot.exe -I boot unix.cma boot/libs.ml boot/duneboot.ml
[  126s] The extension of the output file must be .c, .o or .so
[  126s] make: *** [Makefile:15: dune.exe] Error 2
[  126s] error: Bad exit status from /var/tmp/rpm-tmp.sA7U2Q (%build)
````

(This is essentially `make release`)

--- a/boot/duneboot.ml
+++ b/boot/duneboot.ml
@@ -620,7 +620,7 @@ end = struct
     if ocaml_version < (4, 10) then
       "-custom"
     else
-      "-output-complete-obj"
+      "-output-complete-exe"
 end
 
 let insert_header fn ~header =

